### PR TITLE
Replace Ubuntu 16.04 docker builds with 20.10

### PR DESCRIPTION
**Pull request by kunaltyagi**
In preparation of a 20.10 CI

### CODE REVIEW

File: .ci/azure-pipelines/env.yml [hunk 1]
```diff
@@ -58,6 +58,13 @@ jobs:
         # nvidia-cuda docker image has not been released for 20.04 yet
         USE_CUDA: ""
         tag: 20.04
+      Ubuntu 20.10:
+        CUDA_VERSION: 11
+        UBUNTU_DISTRO: 20.10
+        VTK_VERSION: 8
```
Comment (kunaltyagi) [inline .ci/azure-pipelines/env.yml:64]: Apparently, even 20.10 doesn't have VTK 8 in apt.
```suggestion
        VTK_VERSION: 7
```

File: .ci/azure-pipelines/env.yml [hunk 2]
```diff
@@ -58,6 +58,13 @@ jobs:
         # nvidia-cuda docker image has not been released for 20.04 yet
         USE_CUDA: ""
         tag: 20.04
+      Ubuntu 20.10:
+        CUDA_VERSION: 11
+        UBUNTU_DISTRO: 20.10
+        VTK_VERSION: 7
+        # nvidia-cuda docker image has not been released for 20.04 yet
+        USE_CUDA: ""
+        tag: 20.04
```
Comment (larshg) [inline .ci/azure-pipelines/env.yml:67]: Should this be 20.04?

File: .ci/azure-pipelines/env.yml [hunk 3]
```diff
@@ -58,6 +58,13 @@ jobs:
         # nvidia-cuda docker image has not been released for 20.04 yet
```
Comment (larshg) [inline .ci/azure-pipelines/env.yml:58]: Is not correct anymore - https://hub.docker.com/layers/nvidia/cuda/11.1-devel-ubuntu20.04/images/sha256-c950a0040c0e4c1503834f2ad2ec08d181af72095673c7500dd24a1aacb84893?context=explore

File: .ci/azure-pipelines/env.yml [hunk 4]
```diff
@@ -58,6 +58,13 @@ jobs:
...
         USE_CUDA: ""
         tag: 20.04
+      Ubuntu 20.10:
+        CUDA_VERSION: 11
+        UBUNTU_DISTRO: 20.10
+        VTK_VERSION: 7
+        # nvidia-cuda docker image has not been released for 20.04 yet
```
Comment (larshg) [inline .ci/azure-pipelines/env.yml:65]: Should state 20.10.

**Comment by SunBlack** [comment]
As Nvidia didn't released a CUDA image until now, you could try to run:
```
wget https://developer.download.nvidia.com/compute/cuda/11.1.1/local_installers/cuda_11.1.1_455.32.00_linux.run && \
sudo chmod +x cuda_11.1.1_455.32.00_linux.run && \
sudo sh cuda_11.1.1_455.32.00_linux.run --toolkit --override --silent && \
rm cuda_11.1.1_455.32.00_linux.run
```
On this way I'm installing CUDA on my machines.

**Comment by kunaltyagi** [comment]
Not interested for 20.10

Nvidia will soon resolve the kinks (if any) and release an official release. I have no motivations to support Nvidia ahead of official releases

PS: We need to get 18.04 on `-Werror` ASAP

**Review by mvieth — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Pull request merged**